### PR TITLE
Rename component to Mitsubishi ITP

### PR DESCRIPTION
**Pull request by samphonic**
#### Moved component into climate
```yaml
climate:
  - platform: mitsubishi_itp
    ...
```
#### Changed C namespace to `mitsubishi_itp`

The code itself (and the GitHub site, etc.) will retain MUART references, but the above changes should hide those from end users and protect any future `*_uart` component from name collisions.

### CODE REVIEW

File: tests/components/mitsubishi_itp/common.yaml
Comment (paravoid) [inline tests/components/mitsubishi_itp/common.yaml]: Shouldn't this be `mitsubishi_itp`?

**Comment by paravoid** [comment]
I've been running with this for a few days now, and it's working as expected.

I've reviewed the code and beyond the tiny change in the test above, it LGTM.

I'm still unsure how I feel about keeping the MitsubishiUART references in the code. But at minimum, this PR is a step in the right direction :)

File: tests/components/mitsubishi_itp/common.yaml
Comment (samphonic) [reply tests/components/mitsubishi_itp/common.yaml]: It should!  Thanks.  (I will definitely make sure I run all the tests before merging to dev, but for some reason the test scripts don't run as written in my dev container and I have to modify some files to get them to run.  Minor inconvenience, but then git keeps wanting to push the changes to the test files, etc., etc.)

**Comment by samphonic** [comment]
Thanks for testing it!

Yeah, I'm not the *most* happy with the code still saying MUART all over the place, but it's also going to be a pretty big task to refactor everything (*and* some of the things like the UART Bridge probably should still be called "UART" because that's what they are, so it's not even just a MUART->MITP find and replace).  Plus there's a whole GitHub organization and stuff called "mUART" so I think some of it is just going to stay as like, branding 😉.

Hopefully get all these PRs merged in the next day or so.

**Review by KazWolfe — APPROVED** [review]
LGTM, but I'm curious if having things under the climate entity might cause some interesting edge cases with other sensors/inputs. I guess we'd have noticed if they're broken by now, though.

**Comment by samphonic** [comment]
For some reason I think the additional sensors was the whole reason I went with a top-level component in the first place, but everything seems to be working fine now.  There are other components with similarly embedded sensors though, so I suspect I just didn't understand the YAML well enough when I started to figure it out.

**Pull request merged**
